### ISSUE TITLE
[sklearn#34431] Separate control over response surfaces and class boundaries in `DecisionBoundaryDisplay`

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#326]
**Mirrored from [scikit-learn#34431](https://github.com/scikit-learn/scikit-learn/issues/34431)**

Original title: Separate control over response surfaces and class boundaries in `DecisionBoundaryDisplay`
Original labels: New Feature
Created: 2026-07-03T15:43:11Z

---

### Describe the workflow you want to enable

Originally posted in https://github.com/scikit-learn/scikit-learn/issues/33108#issuecomment-4861934626.

Currently, `DecisionBoundaryDisplay.from_estimator` does not allow users to control whether decision boundaries are drawn or to customize their line style. Although users can customize the surface through `**kwargs`, passing parameters such as `linewidth` or `color` to style the boundary may instead affect the surface and lead to unexpected behavior.

### Describe your proposed solution

I have a suggestion regarding the `DecisionBoundaryDisplay.from_estimator` API. I propose the following interface:

```python
@classmethod
def from_estimator(
    cls,
    estimator,
    X,
    *,
    grid_resolution=100,
    eps=1.0,
    plot_method="contourf",
    response_method="auto",
    class_of_interest=None,
    target_colors=None,
    show_class_boundaries=False,  # new
    response_kwargs=None,         # new
    boundary_kwargs=None,         # new
    xlabel=None,
    ylabel=None,
    ax=None,
):
```

```text
show_class_boundaries : bool, default=False
    Whether to overlay boundaries between predicted classes.

    The boundaries are computed from discrete predictions and are independent
    of the main response displayed by `plot_method`.

response_kwargs : dict, default=None
    Additional keyword arguments passed to the plotting call for the main
    response, such as `alpha`, `levels`, or `zorder`.

boundary_kwargs : dict, default=None
    Additional keyword arguments passed when drawing class boundaries, such as
    `colors`, `linewidths`, `linestyles`, or `zorder`.
```

This API would provide separate and more explicit control over the response surface and the class boundaries, including their colors and other plotting properties.

One open question remains: when `plot_method="contour"` and `response_method="predict"`, `show_class_boundaries` would effectively need to be `True`; otherwise, there would be nothing to display. Perhaps `show_class_boundaries` should instead accept `"auto"`?

### Additional context

Originally discussed in #33108, with additional relevant context in #33015, #32866, and #31553.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/97 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
